opentelemetry/src/common.rs

```diff
@@ -496,7 +493,7 @@ impl InstrumentationScopeBuilder {
     ///     .with_version("v0.1.0")
     ///     .build();
     /// ```
-    pub fn with_version(mut self, version: impl Into<Cow<'static, str>>) -> Self {
+    pub fn with_version(mut self, version: impl Into<String>) -> Self {
```

Comment:
Cow allows us to avoid alloc. Maynot be a concern since scope creation is done at most few times in an app.